### ISSUE TITLE
variable_name rule should extend to all identifiers

### ISSUE THREAD

**Issue by koke** [realm/SwiftLint#663]
The [Swift API Design Guidelines](https://swift.org/documentation/api-design-guidelines/#follow-case-conventions) state that:

> Names of types and protocols are UpperCamelCase. Everything else is
> lowerCamelCase.

The existing `variable_name` rule only checks that variables are lowerCamelCase, but other identifiers, like functions or methods aren't checked:

``` swift
// No error
func Function(param: Int) -> Int {
    return param
}

// error: Variable Name Violation: Variable name should start with a lowercase character: 'Function' (variable_name)
let Function: Int -> Int = { $0 }
```

I'd propose renaming `variable_name` to `identifier_name` and check all the other identifier types as well.

Related: #660

**Comment by ColesyM** [comment]
How do we disable the lower case check in config, for now I have had to disable this entire rule :(
